packages/attestation-service/src/db.ts

```diff
@@ -137,7 +138,32 @@ export async function verifyConfigurationAndGetURL() {
 
 export async function initializeKit(force: boolean = false) {
   if (kit === undefined || force) {
-    kit = newKitFromWeb3(new Web3(fetchEnv('CELO_PROVIDER')))
+    // Prefer passed in keystore if these variables are set
+    const keystoreDirpath = fetchEnvOrDefault('ATTESTATION_SIGNER_KEYSTORE_DIRPATH', '')
+    const keystorePassphrase = fetchEnvOrDefault('ATTESTATION_SIGNER_KEYSTORE_PASSPHRASE', '')
+    if (!keystoreDirpath || !keystorePassphrase) {
+      // Backwards compatibility -- assume this is the deployed full node
+      kit = newKitFromWeb3(new Web3(fetchEnv('CELO_PROVIDER')))
+    } else {
+      const keystoreWalletWrapper = new KeystoreWalletWrapper(new FileKeystore(keystoreDirpath))
+      try {
+        await keystoreWalletWrapper.unlockAccount(
+          fetchEnv('ATTESTATION_SIGNER_ADDRESS'),
+          keystorePassphrase
+        )
+      } catch (error) {
+        kit = undefined
```

Comment:
do you think this fall back to `kit = newKitFromWeb3(new Web3(fetchEnv('CELO_PROVIDER')))`?

Reply:
I think in this case it's best to have it error explicitly (which should happen at startup if the configuration is not correct), otherwise they will hit errors when trying to sign later (since the provider node wouldn't be expected to have the signer unlocked) which could be confusing IMO